src/server/game/Handlers/AuctionHouseHandler.cpp

```diff
@@ -646,6 +646,21 @@ void WorldSession::HandleAuctionListBidderItems(WorldPacket & recvData)
 //this void sends player info about his auctions
 void WorldSession::HandleAuctionListOwnerItems(WorldPacket & recvData)
 {
+    // prevent crash caused by malformed packet
+    uint64 guid;
+    WorldPacket* data = new WorldPacket(recvData);
+    *data >> guid;
+    delete(data);
```

Comment:
IMHO it's not necessary.

You can do it here:

```
    recvData >> guid;
    recvData >> listfrom; 
```

and then you can change the last line of this function from:

` _player->m_Events.AddEvent(new AuctionListOwnerItemsDelayEvent(recvData, _player->GetGUID(), true), _player->m_Events.CalculateTime(delay-diff));`

to 

` _player->m_Events.AddEvent(new AuctionListOwnerItemsDelayEvent(listfrom, guid, true), _player->m_Events.CalculateTime(delay-diff));`

Of course, you have to change the "AuctionListOwnerItemsDelayEvent"  && HandleAuctionListOwnerItemsEvent signature too.

This approach is better for following reasons:

1) you don't have to copy the object and do 2 times the same check
2) the packet must be checked as soon as it's received, is useless to "delay" the check/retrieving of data. The sending of the packet can be delayed safely instead.
3) you won't have the "deleted pointer" issue when passing it to the async function
4) you can keep the async logic :)

Reply:
Done